### PR TITLE
ON-38452 # Added rootFormElementsOnly option to injectable functions

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: src/replaceCustomValues.ts [hunk 1]
```diff
@@ -342,30 +353,38 @@ export function getElementSubmissionValue({
  */
 export function replaceInjectablesWithElementValues(
   text: string,
-  options: {
+  {
+    ...options
+  }: {
```
Comment (Zaxist) [inline src/replaceCustomValues.ts:358]: ?

Comment (mymattcarroll) [reply src/replaceCustomValues.ts:358]: ah, refactor and undo...fixing now.

File: src/form-elements-regex.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  /**
+   * Determine if only root level elements should be matched.
+   *
+   * `false` will match `"{ELEMENT:Parent_Name}"` and
+   * `"{ELEMENT:Children|Name}"`.
+   *
+   * `true` will match `"{ELEMENT:Parent_Name}"` but will NOT replace
+   * `{ELEMENT:Children|Name}`.
+   */
+  rootFormElementsOnly: boolean,
```
Comment (Zaxist) [inline src/form-elements-regex.ts:51]: Obligatory comment about making this `includeNestedElements?: boolean` so getting the existing behaviour is not a breaking change.

Comment (mymattcarroll) [reply src/form-elements-regex.ts:51]: I don't like it but I am on board. Changing now.

Comment (mymattcarroll) [reply src/form-elements-regex.ts:51]: However, to achieve the existing functionality, the variable would need to be `excludeNestedElements`.

**Pull request merged**